### PR TITLE
[systemtest] Remove StatefulSet checks in methods where are not needed

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Refactoring

### Description

This PR removes checks for `StatefulSet`s from STs, which will allow us to execute tests with `StatefulSet` disabled. 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/utils/RollingUpdateUtils.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            Constants.WAIT_FOR_ROLLING_UPDATE_INTERVAL, ResourceOperation.timeoutForPodsOperation(snapshot.size()), () -> {
+                try {
+                    return componentHasRolled(namespaceName, selector, snapshot);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    return false;
+                }
+            });
+
+        LOGGER.info("Component with name: {} has been successfully rolled", componentName);
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/RollingUpdateUtils.java:83]: Maybe label with some proper formating would be useful as well?

**Review by Frawless — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/utils/RollingUpdateUtils.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/RollingUpdateUtils.java:83]: Yeah, that makes sense. Thanks :)

**Review by Frawless — APPROVED** [review]

**Comment by im-konge** [comment]
Test failures are not related to my PR. So this is ready for merge :)

**Pull request merged**